managed/services/backup/pitr_timerange_service.go

```diff
@@ -143,31 +152,42 @@ func (ss *PITRTimerangeService) getPITROplogs(ctx context.Context, location *mod
 	return oplogChunks, nil
 }
 
-func (ss *PITRTimerangeService) ListPITRTimeranges(ctx context.Context, artifactName string, location *models.BackupLocation) ([]Timeline, error) {
+// ListPITRTimeranges returns the available PITR timeranges for the given artifact in the provided location.
+func (s *PBMPITRService) ListPITRTimeranges(ctx context.Context, storage Storage, location *models.BackupLocation, artifact *models.Artifact) ([]Timeline, error) {
 	var timelines [][]Timeline
 
-	oplogs, err := ss.getPITROplogs(ctx, location, artifactName)
+	oplogs, err := s.getPITROplogs(ctx, storage, location, artifact)
 	if err != nil {
 		return nil, errors.Wrap(err, "get slice")
 	}
 	if len(oplogs) == 0 {
 		return nil, nil
 	}
 
-	t, err := gettimelines(oplogs), nil
+	t, err := getTimelines(oplogs), nil
 	if err != nil {
-		return nil, errors.Wrapf(err, "get PITR timeranges for backup '%s'", artifactName)
+		return nil, errors.Wrapf(err, "get PITR timeranges for backup '%s'", artifact.Name)
 	}
 	if len(t) != 0 {
 		timelines = append(timelines, t)
 	}
 
-	return mergeTimelines(timelines...), nil
+	mergedTimelines := mergeTimelines(timelines...)
+	trimTimelines(mergedTimelines)
+
+	return mergedTimelines, nil
+}
+
+// trimTimelines adds one second from Start value of every timeline record. Required to fit PBM values.
+func trimTimelines(timelines []Timeline) {
+	for i, _ := range timelines {
```

Comment:
Right - this problem ), btw if you only care about the index, you can remove the second variable since its optional

```suggestion
	for i := range timelines {
```